ReactAndroid/src/main/res/shell/values/styles.xml

```diff
@@ -10,4 +10,22 @@
         <item name="android:windowFullscreen">true</item>
         <item name="android:windowContentOverlay">@null</item>
     </style>
+
+    <style name="SpinnerDatePickerDialog" parent="Theme.AppCompat.Light.Dialog" tools:targetApi="lollipop">
+        <item name="android:datePickerStyle">@style/SpinnerDatePickerStyle</item>
+    </style>
+
+    <style name="SpinnerDatePickerStyle" parent="android:Widget.Material.Light.DatePicker" tools:targetApi="lollipop">
+        <item name="android:datePickerMode">spinner</item>
+    </style>
+
+
+    <style name="CalendarDatePickerDialog" parent="Theme.AppCompat.Light.Dialog" tools:targetApi="lollipop">
+        <item name="android:datePickerStyle">@style/CalendarDatePickerStyle</item>
+    </style>
+
+    <style name="CalendarDatePickerStyle" parent="android:Widget.Material.Light.DatePicker" tools:targetApi="lollipop">
+        <item name="android:datePickerMode">calendar</item>
+    </style>
```

Comment:
Latest Android API doesn't provide the default methods to change the mode.
Ref: https://developer.android.com/reference/android/widget/DatePicker.html#attr_android:calendarViewShown ,
https://developer.android.com/reference/android/widget/DatePicker.html#attr_android:datePickerMode